tests/popular.controller.test.js

```diff
@@ -0,0 +1,43 @@
+import chai from 'chai';
+import '@babel/polyfill';
+import chaiHttp from 'chai-http';
+import status from '../src/helpers/constants/status.codes';
+// import models from '../src/api/models';
+
+import server from '../src/index';
+
+chai.use(chaiHttp);
+chai.should();
+
+// The test user
+const data = {
+  email: 'alain1@gmail.com',
+  password: 'password23423'
+};
+
+let testUserToken;
+
+describe('popular articles', async () => {
+  before((done) => {
+    chai
+      .request(server)
+      .post('/api/users/login')
+      .send(data)
+      .end((err, res) => {
+        res.should.have.status(status.OK);
+        res.body.should.be.an('Object');
+        testUserToken = res.body.user.token;
+        done();
+      });
+  });
+  it('should get all the popular articles', (done) => {
+    chai
+      .request(server)
+      .get('/api/articles/popular')
```

Comment:
here you are not testing that the expected count/number of popular articles is returned. The PR title says `get 5 mostly read articles` but the tests don't check that:

1.  you're actually getting 5 records and 
2. that the records fetched are actually the most read ones. 

@MCFrank16 we've spoken about proper tests so many times.

Reply:
lemme test it also